lnwallet/channel.go

```diff
@@ -3887,6 +4004,51 @@ func (lc *LightningChannel) validateCommitmentSanity(theirLogCounter,
 			feePerKw, chainfee.FeePerKwFloor)
 	}
 
+	// The channel opener has to account for the commitment fee. This
+	// includes also a buffer type. Depending on whether we are the opener
+	// of the channel we either want to enforce a buffer on the local
+	// amount.
+	var bufferAmt lnwire.MilliSatoshi
+	if lc.channelState.IsInitiator {
+		ourBalance, bufferAmt, err = lc.applyCommitFee(
+			ourBalance, commitWeight, feePerKw, buffer)
+		if err != nil {
+			commitFee := feePerKw.FeeForWeight(commitWeight)
+			lc.log.Errorf("Cannot pay for the CommitmentFee of "+
+				"the ChannelState: ourBalance is negative "+
+				"after applying the fee: ourBalance=%v, "+
+				"commitFee=%v, feeBuffer=%v (type=%v) "+
+				"local_chan_initiator", int64(ourBalance),
+				commitFee, bufferAmt, buffer)
+
+			return err
+		}
+	} else {
+		// No FeeBuffer is enforced when we are not the initiator of
+		// the channel. We cannot do this, because if our peer does not
+		// enforce the FeeBuffer (older LND software) the peer might
```

Comment:
Yeah it will not fix 8315 until we carve-out the cases where we acutally allow the peer dip us below our reserve (outlined in the TODO).

> when it dips below fee buffer, fail the ADD but keep the link

I think this is already in place when this PR is merged, we do not account for the fee buffer when receiving, and when sending we can easily roll back the `ADD`.